### PR TITLE
Cody Web: Add remote context support

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of [SRCH-632](https://linear.app/sourcegraph/issue/SRCH-632/merge-cody-web-experimental-package-to-the-cody-repo-main-branch)

This is the one of PRs that come from the bigger change we did for Cody Web in this main PR https://github.com/sourcegraph/cody/pull/4605

This PR adds support for the remote context items files and symbols. Note that eventually, we will use
open context provides for any remote contexts but for now as a semi-temporary solution for Cody Web 
we include this logic outside the open context providers.

This is safe to merge since it only affects Cody Web package. 

## Test plan
- Check that CI is passing 
- Check cody web demo (`cd web` and `pnpm dev`) and check that you can mention remote files and symbols 
- Check that mentioned files and symbols were included in the prompt message

### CODE REVIEW

File: vscode/src/editor/utils/editor-context.ts [hunk 1]
```diff
@@ -368,9 +417,30 @@ async function resolveFileOrSymbolContextItem(
...
+    if (isRemoteFileURI(contextItem.uri)) {
+        const { repository, path } = parseRemoteFileURI(contextItem.uri)
+
+        // TODO [VK]: Support ranges for symbol context items
+        const resultOrError = await graphqlClient.getFileContent(repository, path)
+
+        if (!isErrorLike(resultOrError)) {
+            return {
+                ...contextItem,
+                uri: URI.from({
```
Comment (abeatrix) [inline vscode/src/editor/utils/editor-context.ts:429]: might have missed something but curious why we want to reset the URI here? Would expect the scheme to be 'remote-file' or wouldn't it be filtered out by SUPPORTED_URI_SCHEMAS later accidentally?

Comment (vovakulikov) [reply vscode/src/editor/utils/editor-context.ts:429]: Good question, so any remote file mentions have a special URI which has "remote-file" scheme, this is how later in places like this handler we determine that this is a remote file and we should use remote file context fetcher logic. 

After this handler, there should be no difference in this file URI compared to other URIs, for example, when we will use this URI to build a file link in the context details in the chat, we should have no special scheme or authority for files like this but common URI with correct path.

**Review by valerybugakov — APPROVED** [review]
LGTM

File: lib/shared/src/mentions/query.ts [hunk 2]
```diff
@@ -31,6 +31,13 @@ export interface MentionQuery {
      * `foo.txt:1`, or `foo.txt:12-`).
      */
     maybeHasRangeSuffix?: boolean
+
```
Comment (thenamankumar) [inline lib/shared/src/mentions/query.ts:34]: I would recommend using the `MentionQuery.provider` field only for this purpose and not introducing this extra field here. 

Given that how we fetch file/symbol results on the web is totally different from that on VS Code, it makes sense not to combine the 2 logic into the current functions to fetch files/symbols from the local context. 


You can set different provider IDs based on mention resolution mode, preferably from inside `parseMentionQuery` only. And if that is not possible then from within the chat context client. 

Then, inside `getChatContextItemsForMention`, you can just add another switch case to capture this and call a separate method to get files/symbol results from remote search.

This won't dirty the current methods fetch the files/symbols and will keep things simple.

File: vscode/src/chat/context/chatContext.ts [hunk 3]
```diff
@@ -37,11 +38,11 @@ export async function getChatContextItemsForMention(
         case SYMBOL_CONTEXT_MENTION_PROVIDER.id:
             telemetryRecorder?.withProvider(mentionQuery.provider)
             // It would be nice if the VS Code symbols API supports cancellation, but it doesn't
-            return getSymbolContextFiles(mentionQuery.text, MAX_RESULTS)
+            return getSymbolContextFiles(mentionQuery.text, MAX_RESULTS, remoteRepositoriesNames)
```
Comment (thenamankumar) [inline vscode/src/chat/context/chatContext.ts:41]: Have separate methods to get context from remote.

File: vscode/src/chat/context/chatContext.ts [hunk 4]
```diff
@@ -37,11 +38,11 @@ export async function getChatContextItemsForMention(
...
         case FILE_CONTEXT_MENTION_PROVIDER.id: {
             telemetryRecorder?.withProvider(mentionQuery.provider)
             const files = mentionQuery.text
-                ? await getFileContextFiles(mentionQuery.text, MAX_RESULTS)
+                ? await getFileContextFiles(mentionQuery.text, MAX_RESULTS, remoteRepositoriesNames)
```
Comment (thenamankumar) [inline vscode/src/chat/context/chatContext.ts:45]: Have separate methods to get context from remote.

File: lib/shared/src/mentions/query.ts (hunk 2 above)
Comment (thenamankumar) [reply lib/shared/src/mentions/query.ts:34]: But if you have already evaluated that idea and it becomes too big of a change to have separate IDs, as we might be mentioning those IDs at a lot of places in the UI as well, then it is fine, you may ignore this.

**Review by thenamankumar — APPROVED** [review]

**Review by umpox — APPROVED** [review]

File: lib/shared/src/mentions/query.ts (hunk 2 above)
Comment (vovakulikov) [reply lib/shared/src/mentions/query.ts:34]: @thenamankumar, thanks for the suggestion; yeah, I was thinking about this and just wanted to make this change as safe as possible and didn't want to change anything about providers. 

As I mentioned in the PR description, I will try to fix open context providers as soon as possible to negate this change in this PR and use open context providers for remote context, so for now I think this change is ok to merge without very deep thinking.

**Pull request merged**